src/scripts/main.js

```diff
@@ -1,3 +1,284 @@
+
+const addNewPerson = (formEvent) => {
+  formEvent.preventDefault();
+
+  const namePerson = document.querySelector("input[data-qa='name']").value;
+  const position = document.querySelector("input[data-qa='position']").value;
+  const office = document.querySelector("select[data-qa='office']").value;
+  const age = +document.querySelector("input[data-qa='age']").value;
+  const salary = +document.querySelector("input[data-qa='salary']").value;
+
+  if (namePerson.length < 4) {
+    pushNotification(
+      'Sorry (-_-)',
+      'The name must be longer than 4 letters',
+      'error',
+    );
+
+    return;
+  }
+
+  if (age < 18 || age > 90) {
+    pushNotification(
+      'Sorry (-_-)',
+      'age must be more than 18 and less than 90',
+      'error',
+    );
+
+    return;
+  }
+
+  tableBody.insertAdjacentHTML('beforeend', `
+  <tr>
+    <td>${namePerson}</td>
+    <td>${position}</td>
+    <td>${office}</td>
+    <td>${age}</td>
+    <td>$${(salary).toLocaleString('en')}</td>
+  </tr>
+`);
+
+  form.reset();
+
+  pushNotification(
+    'Success (^_^)',
+    'New employee was added to table',
+    'success',
+  );
+};
+
+form.addEventListener('submit', addNewPerson);
+
+let topPosition = 20;
+
+function pushNotification(title, text, type) {
+  const div = document.createElement('div');
+  const h2 = document.createElement('h2');
+  const p = document.createElement('p');
+
+  div.classList.add(`${type}`, 'notification');
+  h2.textContent = title;
+  h2.classList.add(`title`);
+  p.textContent = text;
+
+  div.style = `
+  top: ${topPosition}px;
+  right: 20px;
+  `;
+
+  document.body.append(div);
+  div.append(h2);
+  h2.append(p);
+
+  const remove = () => div.remove();
+  const resetTop = () => {
+    topPosition -= 150;
+  };
+
+  topPosition += 150;
+
+  setTimeout(remove, 3000);
+  setTimeout(resetTop, 3000);
+}
+
+let count = false;
+
+const addInput = (mouseEvent) => {
+  if (count) {
+    return;
+  }
+
+  count = true;
+
+  const targetItem = mouseEvent.target.closest('td');
+  const prevText = targetItem.textContent;
+  const isSalaryTarget = prevText.includes('$');
+
+  const input = document.createElement('input');
+
+  input.className = 'cell-input';
+
+  input.value = prevText.replaceAll(/\W/g, '');
+
+  input.type = isNaN(+prevText.replaceAll(/\W/g, ''))
+    ? 'text'
+    : 'number';
+
+  targetItem.textContent = '';
+
+  targetItem.append(input);
+
+  const editValue = (keyboardEvent) => {
+    if (keyboardEvent.key === 'Enter') {
+      const inputValue = document.querySelector('.cell-input').value;
+
+      let text = `${inputValue}`;
+
+      if (input.type === 'number') {
+        text = `$${(+inputValue).toLocaleString('en')}`;
+
+        if (!isSalaryTarget) {
+          text = `${inputValue}`;
+
+          if (+inputValue < 18 || +inputValue > 90) {
+            pushNotification(
+              'Sorry (-_-)',
+              'age must be more than 18 and less than 90',
+              'error',
+            );
```

Comment:
Avoid code duplicate
Imagine that now you hire people under the age of 50... How many times in the code will you have to change this number?

Move all duplicates to helper function or variables